### ISSUE TITLE
Allow for blank entry in password entry dialog

### ISSUE THREAD

**Issue by jeroen** [rstudio/rstudio#4524]
In Rstudio 1.1 the password dialog box forces the user to enter a value. Sometimes we specifically want to allow the user to enter a blank (no value).

```r
askpass <-  getOption('askpass')
askpass("Enter new passphrase. Keep blank for no password")
```

__Update__: in rstudio 1.2 the user is allowed to enter blank, but in this case `askpass` returns `NULL` which is the same value as when the user pressed `cancel` in the password dialog box, or when the IDE failed to show the password prompt. I need to distinguish between the case where "user entered blank password" vs "password dialog failed or canceled".

It would be nice if askpass would return `""` if the user entered no password and pressed `OK`. And still return `NULL` if the user pressed `Cancel`.

**Comment by jeroen** [comment]
Oh wait maybe this is already fixed in rstudio 1.2...

**Comment by ronblum** [comment]
@jeroen Thank you for the suggestion and updated information! We'll consider it as we continue development of RStudio.
